backend/groth16/internal/utils.go

```diff
@@ -0,0 +1,23 @@
+package internal
+
+import "math/big"
+
+// DivideByThresholdOrList divides x into two sub-lists. list must be sorted, non-repeating and contain no value less than indexThreshold
+func DivideByThresholdOrList(indexThreshold int, list []int, x []*big.Int) (ltOrInList, gtAndNotInList []*big.Int) {
```

Comment:
do that in place?

Reply:
Replaced, but it still allocates a slice of size `len(list)`, which in this application is small.